CollAction/Frontend/app/donation/IdealBox.tsx

```diff
@@ -0,0 +1,71 @@
+﻿import renderComponentIf from "../global/renderComponentIf";
+import * as React from "react";
+import * as ReactDOM from 'react-dom';
+import {IdealBankElement} from 'react-stripe-elements';
+
+interface IIdealBoxProps {
+    stripe: any;
+    amount: number;
+    userEmail: string;
+    userName: string;
+}
+
+interface IIdealBoxState {
+    showError: boolean;
+    error: string;
+}
+
+export default class IdealBox extends React.Component<IIdealBoxProps, IIdealBoxState> {
+    constructor(props) {
+        super(props);
+        this.state = {
+            showError: false,
+            error: ""
+        };
+    }
+
+    async submitPayment(event: React.FormEvent<HTMLFormElement>) {
+        event.preventDefault();
+        this.setState({ showError: false });
+
+        let sourceData = {
+            type: "ideal",
+            amount: this.props.amount * 100,
+            currency: "eur",
+            owner: {
+                name: this.props.userName,
+                email: this.props.userEmail
+            },
+            redirect: {
+                return_url: window.location.origin + "/donation/ThankYou"
```

Comment:
What happens if the user cancels the payment? Would be better to return to a dedicated URL that checks if there actually was a payment and then either redirect to the thank-you page or back to the donation page.

Reply:
I don't think there's a way you can cancel iDeal payments, other than clicking the back button or closing the page. My bank doesn't really offer one. If we have sources that could be cancelled, this would make sense. I initially had a more generic return url, but it only makes sense with more payment options.

Reply:
I'll add it back anyway, we can't test all banks, and it might change in the future.

Reply:
Did that, and fixed a bug in the payment flow (people could complete the iDeal payment, but we never "charged" the iDeal source).